### DISCUSSION TITLE
Reconcile method for SyncVars

### DISCUSSION THREAD

**Discussion by AlphaLul** [FirstGearGames/FishNet#676] [Feature Request]
Currently, server-authoritative predicted SyncVars have the client tell the server what to set it to and the server does so. If the value is invalid and you want to reconcile, you have to invoke a TargetRpc to correct it on the owner client. It would be a much easier experience if there was a SyncVar method to tell the owner client to reconcile a predicted SyncVar instead of having to make a TargetRpc for every variable.

Below are examples of the current workflow and what I imagine the new workflow would be like to display the benefits. The difference is small with only 1 SyncVar, but the TargetRpcs add up fast if you have multiple variables.

Current workflow:
```csharp
//Number should only be 1 through 5
private readonly SyncVar<int> number = new SyncVar<int>(new SyncTypeSettings(WritePermission.ClientUnsynchronized, ReadPermission.ExcludeOwner));
//Is called by the client somewhere
[ServerRpc(RunLocally = true)]
private void SetNumber(int value, NetworkConnection conn = null)
{
    number.Value = value;
        
    if (!IsServerStarted) return;
    //Limit to range of 1-5, set to 1 if outside range
    if (value < 1 || value > 5)
    {
        number.Value = 1;
        ReconcileNumber(conn, number.Value);
    }
}
    
[TargetRpc]
private void ReconcileNumber(NetworkConnection conn, int newNumber)
{
    number.Value = newNumber;
}
```

Proposed workflow:
```csharp
//Number should only be 1 through 5
private readonly SyncVar<int> number = new SyncVar<int>(/*May need new SyncTypeSetting for this*/);
//Is called by the client somewhere
[ServerRpc(RunLocally = true)]
private void SetNumber(int value)
{
    number.Value = value;
        
    if (!IsServerStarted) return;
    //Limit to range of 1-5, set to 1 if outside range
    if (value < 1 || value > 5)
    {
        //Call var.Reconcile to set the correct value on the server and instruct the client to do the same, removing the need for an extra TargetRpc for each variable
        number.Reconcile(1);
    }
}
```

**Comment by FirstGearGames** [comment]
SyncVars are outside the prediction loop. If anything we're going to move towards two different work flow rather than try to mash everything together.
